patches/chromium/fix_software_compositing_infinite_loop.patch

```diff
@@ -0,0 +1,31 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Samuel Maddock <smaddock@slack-corp.com>
+Date: Fri, 18 Oct 2024 11:11:11 -0400
+Subject: fix: software compositing infinite loop
+
+When GPU compositing is unavailable, LayerTreeView::RequestNewLayerTreeFrameSink
+may run in an infinite loop due to a race condition. Need to allow time to
+process CompositingModeFallbackToSoftware IPC to disable GPU compositing.
+
+https://issues.chromium.org/345275130
+
+diff --git a/third_party/blink/renderer/platform/widget/compositing/layer_tree_view.cc b/third_party/blink/renderer/platform/widget/compositing/layer_tree_view.cc
+index 9497ab591864295231db47fdc526e59935e6aa31..635fa41649db41800f99da2683cc955b2e1e935f 100644
+--- a/third_party/blink/renderer/platform/widget/compositing/layer_tree_view.cc
++++ b/third_party/blink/renderer/platform/widget/compositing/layer_tree_view.cc
+@@ -374,9 +374,13 @@ void LayerTreeView::DidFailToInitializeLayerTreeFrameSink() {
+   // unable to be killed after Chrome is closed.
+   // https://issues.chromium.org/336164423
+   if (!Platform::Current()->IsGpuRemoteDisconnected()) {
+-    layer_tree_host_->GetTaskRunnerProvider()->MainThreadTaskRunner()->PostTask(
++    // CompositingModeFallbackToSoftware IPC will disable GPU compositing in
++    // RenderThread. Post task with delay to give time to receive this IPC and
++    // prevent infinite loop of retries for software renderers.
++    // https://issues.chromium.org/345275130
++    layer_tree_host_->GetTaskRunnerProvider()->MainThreadTaskRunner()->PostDelayedTask(
+         FROM_HERE, base::BindOnce(&LayerTreeView::RequestNewLayerTreeFrameSink,
+-                                  weak_factory_.GetWeakPtr()));
++                                  weak_factory_.GetWeakPtr()), base::Milliseconds(10));
```

Comment:
The [docs on `SequencedTaskRunner`](https://source.chromium.org/chromium/chromium/src/+/main:base/task/sequenced_task_runner.h;l=127;drc=04989307509c78d67b34f2282ca549397719d046) make me think that this delay could potentially be as low as 1 nanosecond and still give the other task priority. Not sure though.

Reply:
I tested with 1ms delay which seems to also fix the issue. Throttling the timer may reduce CPU usage slightly in cases where it fails multiple times in a row.

When set to 10ms, I saw `LayerTreeView::DidFailToInitializeLayerTreeFrameSink` posted 4 times. When set to 1ms, I only saw 2 occurrences. I'll leave it up to you when creating the CL upstream. The Chromium folks may also have thoughts as to what an appropriate delay might be.